pkg/healthcheck/healthcheck_test.go

```diff
@@ -2124,6 +2171,93 @@ metadata:
 	})
 }
 
+func getConfigAndKubeSystemNamespace(ha bool, nsLabel string) []string {
+	return []string{fmt.Sprintf(`
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: linkerd-config
+  namespace: linkerd
+data:
+  install: |
+    {"cliVersion":"dev-undefined","flags":[{"name":"ha","value":"%v"}]}`, ha),
+		fmt.Sprintf(`
+apiVersion: v1
+kind: Namespace
+metadata:
+  creationTimestamp: null
+  labels:
+    %s
+    linkerd.io/is-control-plane: "true"
+  name: kube-system`, nsLabel),
+	}
+}
+
+func TestKubeSystemNamespaceInHA(t *testing.T) {
+	testCases := []struct {
+		testDescription string
+		k8sConfigs      []string
+		expectedOutput  string
+	}{
+		{
+			"passes when HA is not enabled",
+			getConfigAndKubeSystemNamespace(false, ""),
+			"",
+		},
+		{
+			"passes when HA is enabled and namespace has required metadata",
+			getConfigAndKubeSystemNamespace(true, "config.linkerd.io/admission-webhooks: disabled"),
+			"l5d-injection-disabled pod injection disabled on kube-system",
+		},
+		{
+			"fails when HA and admission hooks are enabled",
+			getConfigAndKubeSystemNamespace(true, "config.linkerd.io/admission-webhooks: enabled"),
+			"l5d-injection-disabled pod injection disabled on kube-system: kube-system namespace needs to have config.linkerd.io/admission-webhooks: disabled if HA mode is enabled",
```

Comment:
I would precise here that it needs to have the *label* `config.linkerd.io/admission-webhooks: disabled`